cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
 import importlib
 import importlib.util
 import inspect
+import os
+import subprocess
 import sys
 from pathlib import Path
 
 import click
 import typer
 
+from cognite.dm_clients.config import settings
 from cognite.dm_clients.domain_modeling.schema import Schema
 from cognite.gqlpygen.generator import to_client_sdk
 from cognite.gqlpygen.misc import to_client_name, to_schema_name
 
 app = typer.Typer()
 
 
-@app.command("topython", help="Input a .graphql schema to create pydantic schema.")
+_settings_cognite = settings.get("cognite", {})
+_cdf_cluster = _settings_cognite.get("cdf_cluster")
+_tenant_id = _settings_cognite.get("tenant_id")
+_client_id = _settings_cognite.get("client_id")
+_client_secret = _settings_cognite.get("client_secret")
+_project = _settings_cognite.get("project")
+
+_settings_dm_clients = settings.get("dm_clients", {})
+_space = _settings_dm_clients.get("space")
+_datamodel = _settings_dm_clients.get("datamodel")
+_schema_version = _settings_dm_clients.get("schema_version")
+
+_settings_local = settings.get("local", {})
+_prefix = _settings_local.get("prefix", "")
+_graphql_schema = _settings_local.get("graphql_schema")
+_schema_module = _settings_local.get("schema_module")
+
+_schema_dir = Path(_graphql_schema).parent if _graphql_schema else None
+
+_client_secret_none = "None (use device flow)"
+if _client_secret:
+    _client_secret_hidden = f"{_client_secret[:3]}{'*' * (len(_client_secret) - 3)}"
+else:
+    _client_secret_hidden = _client_secret_none
+
+
+@app.command("topython", help="Create pydantic schema and Python DM client from a .graphql schema.")
 def to_python(
-    graphql_schema: Path = typer.Argument(..., help="GraphQL schema to convert"),
-    name: str = typer.Option(
-        "MySchema", "--name", help="Name of the client and schema, expected to be in pascal case."
-    ),
+    graphql_schema: Path = typer.Argument(_graphql_schema or ..., help="GraphQL schema to convert"),
+    output_dir: Path = typer.Option(_schema_dir or os.getcwd(), help="Directory to write schema.py and client.py to."),
+    prefix: str = typer.Option(_prefix, help="Name prefix for the domain."),
```

Comment:
Not sure if I agree with this one. I know it is technically a prefix, but conceptually it is a name for the domain.

Reply:
Yeah, I'm in two minds about it. Thing is, the name / prefix is not required. That makes it more of a prefix then a name. But maybe name has more clear meaning.

Reply:
After sleeping on it, I prefer "prefix" because "name" is a more loosely-defined term.No strong feelings about it, though. You decide: PR https://github.com/cognitedata/cognite-gql-pygen/pull/22 :)